### PR TITLE
#21212: Fix softmax_in_place golden func

### PR DESCRIPTION
**Pull request by skrsticTT**
### Ticket
#21212

### Problem description
The `ttnn.softmax_in_place` function from SD 1.4 has PCC issues when comparison mode is enabled.

### What's changed
In softmax calls, dim defaults to -1 if not specified. Do the same for the golden implementation. Add a test that references this golden implementation too.

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/runs/16799059623) tt-train tests are failing on the main, unrelated issue

### CODE REVIEW

**Comment by skrsticTT** [comment]
fyi @pavlejosipovic

**Review by bbradelTT — APPROVED** [review]

File: ttnn/ttnn/operations/normalization.py [hunk 1]
```diff
@@ -20,6 +20,8 @@ def find_closest_largest_divisor(num: int, start_divisor: int):
 def _golden_function(input_tensor: ttnn.Tensor, dim: Optional[int] = None, **_):
     import torch
 
+    dim = dim or -1
```
Comment (omilyutin-tt) [inline ttnn/ttnn/operations/normalization.py:23]: Is it possible to align TT-NN default with pytorch?

**Review by omilyutin-tt — APPROVED** [review]

**Review by ayerofieiev-tt — APPROVED** [review]

File: ttnn/ttnn/operations/normalization.py (hunk 1 above)
Comment (skrsticTT) [reply ttnn/ttnn/operations/normalization.py:23]: As far as I know, pytorch does not have a default value for dim; it must be explicitly specified. @bbradelTT, can you confirm?

Comment (skrsticTT) [reply ttnn/ttnn/operations/normalization.py:23]: But I suppose that (if needed) it should be in a separate PR

Comment (bbradelTT) [reply ttnn/ttnn/operations/normalization.py:23]: Looks like they deprecated the functionality, but currently there's only a warning:
```
>>> import torch
>>> a = torch.nn.functional.softmax(torch.ones([1,1]))
<stdin>:1: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
>>> a
tensor([[1.]])
```
